elytron/WFCORE-5027-charset-encoding-support.adoc

```diff
@@ -0,0 +1,236 @@
+
+The ``UsersRolesLoginModule``, ``DatabaseServerLoginModules`` and ``LDAPLoginModule`` share
+``UsernamePasswordLoginModule`` as their parent module. Ergo, they all support configuring the hashing algorithm, encoding
+and character set to be used across all passwords in the module.
+
+The Elytron security realms provide similar functionality than the descendants of the
+``UserRolesLoginModule``. For instance, the Properties Security Realm uses a properties file
+similarly to the ``UsersRolesLoginModule``. Furthermore, the ``DatabaseServerLoginModule``
+is a JDBC login module that supports authentication and role mapping, whose behaviour
+is replicated by the JDBC realm in the Elytron subsystem. Finally, the ``LDAPLoginModule`` authenticates
+against an LDAP server just like the LDAP security realm in Elytron.
+
+=== PicketBox Authentication
+
+Given a hashing algorithm has been provided as a module option, the Legacy modules https://github.com/picketbox/picketbox/blob/master/security-jboss-sx/jbosssx/src/main/java/org/jboss/security/auth/spi/UsernamePasswordLoginModule.java#L263[hash] the client provided passwords prior to validating whether it matches
+the store password. Note that validating whether the store password and the client provided password  might have a module specific
+implementation i.e. in the https://github.com/picketbox/picketbox/blob/master/security-jboss-sx/jbosssx/src/main/java/org/jboss/security/auth/spi/LdapLoginModule.java#L265[LDAPLoginModule] it takes the client provided password and it tries to
+create an  LDAP init context with the client password and username.
+
+All modules use the https://github.com/picketbox/picketbox/blob/master/security-jboss-sx/jbosssx/src/main/java/org/jboss/security/auth/spi/UsernamePasswordLoginModule.java#L190[UsernamePasswordLoginModule#login] method to perform the authentication
+which creates an identity  based on the username provided by the client,
+retrieves the password in the store corresponding to that identity, hashes the client provided password
+and checks whether the hashed client password matches the password in the store.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFCORE-5027[WFCORE-5027]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY-13625[WFLY-13625]
+* https://issues.redhat.com/browse/ELY-2001[ELY-2001]
+* https://issues.redhat.com/browse/EAP7-1104[EAP7-1104]
+
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+TBD
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [ ] QE
+
+TBD
+
+=== Affected Projects or Components
+
+* Wildfly, Wildfly-Core and Elytron
+
+=== Other Interested Projects
+
+N/A
+
+== Requirements
+
+=== Hard Requirements
+
+All realms that support hashed passwords need to be modified: Properties Realm, Filesystem Realm, JDBC Realm, and LDAP Realm.
```

Comment:
Would be good to add links to the corresponding Java classes.

Reply:
Added!